consensus/spos/worker.go

```diff
@@ -316,22 +317,27 @@ func (wrk *Worker) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedP
 		return err
 	}
 
-	topic := GetConsensusTopicIDFromShardCoordinator(wrk.shardCoordinator)
+	topic := GetConsensusTopicID(wrk.shardCoordinator)
 	err = wrk.antifloodHandler.CanProcessMessagesOnTopic(message.Peer(), topic, 1, uint64(len(message.Data())), message.SeqNo())
 	if err != nil {
 		return err
 	}
 
+	defer func() {
+		shouldBlacklistPeer := err != nil && !errors.Is(err, ErrMessageForPastRound)
```

Comment:
I think it would work without the first check err != nil - only the second one should be enough

Reply:
Its not working. For err == nil the method will return false and the shouldBlacklistPeer will be set wrongly to true. Added an unit test to check this.